src/west/manifest-schema.yml

```diff
@@ -133,3 +142,6 @@ mapping:
               clone-depth:
                 required: false
                 type: int
+              is-zephyr-base:
```

Comment:
I think we could generalize this. I am thinking how best to do this but how about something like:

```
env-map:
   required: false
   type: str
```

So then you could do things like:

```
  projects:
    - name: zephyr
      env-map: ZEPHYR_BASE
    - name: socs
      env-map: SOC_DIR
    - name: boards
      env-map: BOARDS_DIR
```

Reply:
I tried to think about a general solution but I wasn't sure I had a "good enough" answer. Coming up with an ad-hoc hack we can deprecate later feels better than coming up with a general solution that doesn't quite fit.

I can see issues that make this potentially overly limiting:

- what if i wanted to set two or more environment variables based on some attribute of the project? (having the key be a string like ZEPHYR_BASE instead of a sequence or other compound data type means we can only get one value)
- what if i want to set a relative path instead of an absolute path?
- what if i wanted to append onto an existing environment variable (like PATH, PYTHONPATH) instead? (perhaps with a or platform-dependent configurable separator)
- what if i want to use a subpath inside a project? ("append this-project-abspath/bin to PATH")

Reply:
> Coming up with an ad-hoc hack we can deprecate later feels better than coming up with a general solution that doesn't quite fit.

... so one potential solution is "use the project named zephyr if it exists, or the project with path zephyr if that fails" and we can defer trying to be clever.

Reply:
> I tried to think about a general solution but I wasn't sure I had a "good enough" answer. Coming up with an ad-hoc hack we can deprecate later feels better than coming up with a general solution that doesn't quite fit.
> 
> I can see issues that make this potentially overly limiting:
> 
> * what if i wanted to set two or more environment variables based on some attribute of the project? (having the key^W edit: value be a string like ZEPHYR_BASE instead of a sequence or other compound data type means we can only get one value)
> * what if i want to set a relative path instead of an absolute path?
> * what if i wanted to append onto an existing environment variable (like PATH, PYTHONPATH) instead? (perhaps with a or platform-dependent configurable separator)
> * what if i want to use a subpath inside a project? ("append this-project-abspath/bin to PATH")

I see your point, but wouldn't all of those be solvable by adding further sub-options in the manifest later on? i.e. instead of going from the bottom up, we could go top-down by adding a generic environment variable setting in the manifest that can later be refined by adding to the schema in a way that we cover all the variants that you mention.

Reply:
> > Coming up with an ad-hoc hack we can deprecate later feels better than coming up with a general solution that doesn't quite fit.
> 
> ... so one potential solution is "use the project named zephyr if it exists, or the project with path zephyr if that fails" and we can defer trying to be clever.

Actually going back to it now, I think we don't know enough at this point about the requirements and about the exact way we're gonna split Zephyr to take a decision, so I agree with your assessment above and would like to propose deferring the decision as well.

Reply:
> I see your point, but wouldn't all of those be solvable by adding further sub-options in the manifest later on?

Not unless you want to deprecate the env-map proposal, or keep it but add other keys on the side which do "env-map-and-more". The pykwalify schema wants the values to be of a certain type, which we would have to change incompatibly to make the additions to `env-map` I am asking about.